cli_tree.md

```diff
@@ -0,0 +1,102 @@
+# CLI graph/tree handling
+
+This document scopes the extension of the OMERO Command Line Interface to
+have a better support for trees and graphs.
+
+## Context
+
+We already introduced the CLI [obj plugin](http://www.openmicroscopy.org/site/support/omero5.2/developers/cli/obj.html) to manipulate individual objects. The generic syntax of this plugin is of
+the form:
+
+    bin/omero obj subcommand  Object:id field1=value1 field2=value2 ...
+
+We are facing more use cases where we need to manipulate graphs and sets of 
+objects. A first implementation of such plugin has been provided in the context
+of IDR with the [CLI children plugin](https://github.com/openmicroscopy/openmicroscopy/pull/4182).
+
+The current consensus is to preserve the `obj` plugin for the manipulation of
+individual objects and design a new plugin for graph/set tasks. For now, we will use the term `tree` to refer to this new plugin.
+
+## Tree plugin
+
+This plugin could include a set of subcommands. Each subcommand could either
+be a read operation applied to a graph specified by the `graph-spec` argument:
+
+    bin/omero tree read-operation graph-spec
+
+or a write operations applied to a `graph-spec` argument with an optional
+`object-spec` target:
+
+    bin/omero tree write-operation graph-spec [obj-spec]
+
+## Input specification
+
+There are 2 proposed formats that could be supported when specifying the input
+`graph-spec`:
+
+-   either stay close to the current `obj-spec` format, i.e. `Object:id`
+-   or move towards xpath-like formats e.g. `//Image/*`, `Image[name=xxx]`, `Screen:id/*/Roi:*`
+
+If sticking to the first option, the proposed representation of orphaned
+objects would be `Image:orphan`.
+
+## Tree subcommands
+
+Below is a list of proposed subcommands with use cases.
+
+### Link
+
+Move orphaned images into a dataset:
+
+      omero tree link Image:orphan Dataset:1
+      omero tree link //Image:* Dataset:1
+
+Link images in between datasets
+
+    omero tree link Dataset:1 Dataset:2
```

Comment:
Here the xpath style reads much better than the older style. While there can be only images in a Dataset this command suggests the linking of the two Datasets rather than the conents of one. In this case I prefer the explicitness of the xpath format. Though from an example of `tree copy` below I assume it could also be expressed as:

```
omero tree link Dataset:1/* Dataset:2
```